### DISCUSSION TITLE
Avoid dataloaders with this idea?

### DISCUSSION THREAD

**Discussion by frederikhors** [99designs/gqlgen#1928] [Ideas]
Hi guys.

What do you think about implementing [this idea](https://blog.logrocket.com/designing-graphql-server-optimal-performance/) in gqlgen?

It can be amazing and we can usefully avoid dataloaders.

**Comment by mattlysf** [comment]
This seems to add a lot of unnecessary complexity to resolving leaf nodes. You can always have a dataloader behind a struct method if you want it to be deferred. 

If dataloaders aren't your thing, you can also avoid n+1 with non-eager preloading by looking at the selections /  field collection: https://gqlgen.com/reference/field-collection/
